adapters_l1_attributes.go

```diff
@@ -0,0 +1,45 @@
+package monomer
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+// global variable to avoid creating it every time
+var (
+	L1AttributesTx = ethtypes.DepositTx{
+		// From:                derive.L1InfoDepositerAddress,
+		To:                  &derive.L1BlockAddress,
+		Mint:                nil,
+		Value:               big.NewInt(0),
+		Gas:                 derive.RegolithSystemTxGas,
+		IsSystemTransaction: false,
+	}
+	MethodID = derive.L1InfoFuncEcotoneBytes4
+)
+
+// L1 Attributes Deposited Transaction
+// https://specs.optimism.io/protocol/deposits.html#l1-attributes-deposited-transaction
+// After Ecotone update we no longer need to check calldata
+//nolint:lll
+// https://github.com/ethereum-optimism/optimism/blob/7e5b9faa14d1f0a8d3cf919654162c4a402f1f38/packages/contracts-bedrock/src/L2/L1Block.sol#L125C44-L125C56
+
+func IsL1AttributesTx(tx *ethtypes.Transaction) bool {
+	// tx.To() != nil && tx.To().Cmp(*L1AttributesTx.To) == 0 - two calls to To()
+	if to := tx.To(); to == nil || to.Cmp(*L1AttributesTx.To) != 0 {
+		return false
+	}
+	if value := tx.Value(); value == nil || value.Cmp(L1AttributesTx.Value) != 0 {
+		return false
+	}
+	if txData := tx.Data(); txData == nil || len(txData) <= len(MethodID) || !bytes.Equal(txData[:len(MethodID)], MethodID) {
+		return false
+	}
+
+	// TODO: Can we check From field?
+
+	return tx.Gas() == L1AttributesTx.Gas && tx.IsSystemTx() == L1AttributesTx.IsSystemTransaction && tx.Mint() == nil
```

Comment:
Line is a mouthful, and the checks are each slightly different as well

```suggestion
	return tx.Gas() == L1AttributesTx.Gas &&
	 tx.IsSystemTx() == L1AttributesTx.IsSystemTransaction &&
	 tx.Mint() == nil
```